newIDE/app/src/SceneEditor/index.js

```diff
@@ -876,6 +1008,33 @@ export default class SceneEditor extends React.Component<Props, State> {
     this.deleteSelection();
   };
 
+  duplicateSelection = () => {
+    const serializedSelection = this.instancesSelection
+      .getSelectedInstances()
+      .map(instance => serializeToJSObject(instance));
+
+    if (!this.editor) return;
+
+    const newInstances = serializedSelection
+      .map(serializedInstance => {
+        const instance = new gd.InitialInstance();
+        unserializeFromJSObject(instance, serializedInstance);
+        return instance;
+      })
+      .map(instance => {
```

Comment:
Do you need to do two maps? If you do a single one, this allows to keep the new in the same function as the `delete`, which is looking a tiny bit safer for the reader.

Reply:
I believe you can also change the `paste` function to remove the two maps yeah, it seems useless..